### PR TITLE
Add ERC: Decentralized Employment System

### PR DESCRIPTION
**Pull request by jamesavechives**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: ERCS/erc-0000.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+eip: 0000
```
Comment (abcoathup) [inline ERCS/erc-0000.md:2]:
```suggestion
eip: 7750
```
Assigning next sequential EIP/ERC/RIP number.

Please also update the filename.

File: ERCS/erc-0000.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+title: Decentralized Employment System
+description: A blockchain-based employment system that records employment history.
+author: James Savechives (@jamesavechives) <james.walstonn@gmail.com>
+discussions-to: https://ethereum-magicians.org/t/discussion-on-eip-xxx-decentralized-employment-system-des/20578
```
Comment (abcoathup) [inline ERCS/erc-0000.md:6]: Please create a topic in Eth Magicians, it can just contain a link to this PR.

**Comment by Tara-annison** [comment]
Just my view but this feels more like an app layer design/implementation than a core protocol design/implementation.

File: ERCS/erc-7750.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+---
+eip: 7750
+title: Decentralized Employment System
+description: A blockchain-based employment system that records employment history.
```
Comment (SamWilsn) [inline ERCS/erc-7750.md:4]:
```suggestion
description: An employment system that records employment history.
```

Almost all ERCs are blockchain-based, so mentioning it here takes up valuable space in your description. You should use more of the available space here to describe your proposal.

File: ERCS/erc-7750.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+### Firing Employees
+
+1. **Termination Conditions**: The contract specifies conditions under which the company can terminate an employee. These conditions are enforced by the smart contract.
+2. **Dispute Resolution**: A decentralized dispute resolution mechanism can be integrated to handle disagreements between companies and employees.
+
+### Review System
+
+1. **Reputation Scores**: Both companies and employees can receive and give reviews. These reviews are stored on-chain and contribute to a reputation score.
+2. **Public Reviews**: Reviews are public and accessible to all users, fostering accountability and trust in the system.
```
Comment (SamWilsn) [inline ERCS/erc-7750.md:52]: You're kind of describing your desired outcome without giving any specific technical instructions to implementers. How will someone know how to represent a review stored on-chain? What functions need to be called? What contracts does the system use?

This is way too vague, even for a draft.

File: ERCS/erc-7750.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+## Rationale
+
+The DES is designed to create a fair, transparent, and secure employment ecosystem. By recording employment history on-chain, automating contract enforcement, and providing a public review system, DES addresses many of the shortcomings of traditional employment systems.
```
Comment (SamWilsn) [inline ERCS/erc-7750.md:56]: You're justifying the entire proposal here. This section should, instead, be used to explain technical choices made within the document itself.

File: ERCS/erc-7750.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+
+## Test Cases
+
+1. **Company Creation**: Test the registration and profile creation for new companies.
+2. **Contract Creation and Signing**: Test the creation of a labor contract and the signing process by both parties.
+3. **Salary Payment**: Test the salary deposit and automated payment system.
+4. **Employment Termination**: Test the conditions for employee termination and the associated smart contract enforcement.
```
Comment (SamWilsn) [inline ERCS/erc-7750.md:64]: Test cases should be a simple set of inputs and expected outputs/states.

File: ERCS/erc-7750.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+## Copyright
+
+Copyright and related rights waived via [CC0](../LICENSE.md).
+
+---
+
+### About the Author
+
+James Savechives is a blockchain enthusiast and developer with a keen interest in digital assets and their innovative applications. With a background in software development, James is currently working on the Aizel Network, aiming to explore and create solutions that bridge the gap between technology and practical applications.
```
Comment (SamWilsn) [inline ERCS/erc-7750.md:80]:
```suggestion
```

This isn't relevant to the specification.

**Review by SamWilsn — CHANGES_REQUESTED** [review]

File: ERCS/erc-7750.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+### Firing Employees
+1. **Termination Conditions**: The `terminateContract(uint contractId, string reason)` function can be called by the company if the termination conditions outlined in the contract are met. The function sets the contract status to "terminated."
+2. **Moderator's Role in Termination**: In the case of disputes regarding termination, the moderator is called upon to resolve the issue based on the contract terms and evidence provided by both parties.
+
+### Review System
+1. **Reputation Scores**: Both companies and employees can submit reviews using the `submitReview(uint contractId, uint rating, string comments)` function. These reviews are stored in a `reviews` mapping and tied to the contract ID.
+2. **Public Reviews**: Reviews can be accessed using `getReviews(uint contractId)`, which returns an array of reviews associated with a specific contract. These reviews contribute to a reputation score for both the company and employee.
```
Comment (SamWilsn) [inline ERCS/erc-7750.md:54]: Generally ERCs propose a complete solidity interface, then explain the function/behaviour of its functions. While it isn't a strict requirement, I'd like to see it here.

File: ERCS/erc-7750.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+It must be noted that this system comes into play after the company tests the employee, the employee passes these tests, and before hiring and signing the labor contract. The employee uses their employee token, which contains their employment history, and the company checks the history before finalizing the labor contract.
+
+## Motivation
+Traditional employment systems are centralized, opaque, and often lack trust. The DES aims to bring transparency, immutability, and trust to employment by leveraging blockchain technology. By recording employment history on-chain, enabling decentralized company creation, automating contract enforcement, and adding moderation for dispute resolution, the DES promotes a fairer and more transparent employment ecosystem. This system also streamlines the final steps of hiring, following testing phases, making the contract phase smoother and more secure.
+
+## Specification
+
+### Company Creation
+1. **Company Registration**: Users can register new companies on the blockchain. Each company is a unique entity with its own address.
+2. **Company Profile**: The company profile includes details like name, industry, and a list of registered employees.
```
Comment (SamWilsn) [inline ERCS/erc-7750.md:25]: These requirements are too vague. How does a user register a new company? How is the profile stored/retrieved? What's the API for these actions?

File: ERCS/erc-7750.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+### Labor Contracts
+1. **Contract Creation**: Once the company has tested the employee and is ready to move forward with the final hiring stage, companies can create a labor contract by calling the `createContract(uint companyId, uint employeeTokenId, uint salary, uint duration, string responsibilities, string terminationConditions)` function. This function generates a unique `contractId` and stores it in a `contracts` mapping, associating each contract with a `companyId`.
```
Comment (SamWilsn) [inline ERCS/erc-7750.md:28]: In which contract is the `createContract` function defined?

Also, side note, but calling it a contract may be confusing unless an employment contract is also a smart contract. Maybe employment record?

Comment (SamWilsn) [inline ERCS/erc-7750.md:28]: > stores it in a `contracts` mapping, associating each contract with a `companyId`

This is an implementation detail. ERCs should only specify the externally visible behaviour of a contract, not how the contract is implemented.

File: ERCS/erc-7750.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+   - The employee will handle their **employee token** (minted prior to employment) to the company, and the company will review the token's employment history before proceeding with the contract.
```
Comment (SamWilsn) [inline ERCS/erc-7750.md:29]: What kind of token is this? ERC-721? A soulbound token?

**Review by SamWilsn — COMMENTED** [review]
The biggest blocker for getting this one merged is that the specification section is too informal and lacking in detail.

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**